### PR TITLE
Fetch all coins and individual coin page 2nd try

### PR DESCRIPTION
**Pull request by Alcladii**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ import CoinPage from "./pages/CoinPage"
 import queryString from 'query-string';
 import {SearchItemInput} from './components/SearchInput'
 import { ResultList } from "./components/ResultList";
+import {CurrencySelector} from "./components/CurrencySelector"
+import {CryptoProvider} from "./contexts/cryptoContext"
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:15]: add spaces between {}

File: src/App.jsx [hunk 2]
```diff
@@ -20,10 +22,12 @@ let urlObject = {
 
 
 export default function App() {
+
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:25]: n new line

File: src/App.jsx [hunk 3]
```diff
@@ -38,11 +42,14 @@ export default function App() {
...
-            </div>            
+            </div>  
+            <div>
+              <CurrencySelector />
+            </div>          
           </div>               
         </nav>
 
-        <Switch>
+        <Switch>
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:52]: remove extra space

File: src/App.jsx [hunk 4]
```diff
@@ -53,7 +60,9 @@ export default function App() {
             <CoinPage {...props}/>
```
Comment (yoloOnTheBattlefield) [inline src/App.jsx:60]: looks into link and fetching data via url params like in the weather app

File: src/contexts/cryptoContext.jsx [hunk 5]
```diff
@@ -12,4 +13,80 @@ const useLocalState = (key, initialValue) => {
...
+      /*setIsLoading(true);
+      const {data}  = await axios(      
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${coinPage}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+      );
+      setCoinList(coinList.concat(data))
+      setCoinPage(coinPage + 1)*/
+      /*const { coins }  = await api(
+        "/coins/markets",
+        "vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d"
+      );*/
```
Comment (yoloOnTheBattlefield) [inline src/contexts/cryptoContext.jsx:36]: const order = coinListDsc ? "market_cap_desc" : "market_cap_asc"
setIsLoading(true)
        const response = await api(
          "/coins/markets",
          "vs_currency=usd&order=${order}&per_page=50&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d"
        );

File: src/contexts/cryptoContext.jsx [hunk 6]
```diff
@@ -12,4 +13,80 @@ const useLocalState = (key, initialValue) => {
...
+      } catch (err) {
+      //console.log("Error in Fetching Data")
+      setHasError(true); 
+      setisLoading(false);
+    }
+  };
+
+  const setToDsc = () => {
+    setCoinListDsc(true); 
+    console.log("setToDsc run")
```
Comment (yoloOnTheBattlefield) [inline src/contexts/cryptoContext.jsx:69]: no console

File: src/contexts/cryptoContext.jsx [hunk 7]
```diff
@@ -12,4 +13,80 @@ const useLocalState = (key, initialValue) => {
...
+        coins = response.data;   
+        //console.log("bottom 50", coins)
+      }
+      //console.log(coins)
+      setCoinList(coins);
+      setIsLoading(false);
+      setHasError(false)
+      //console.log(data)
+      } catch (err) {
+      //console.log("Error in Fetching Data")
```
Comment (yoloOnTheBattlefield) [inline src/contexts/cryptoContext.jsx:61]: no commented out code

File: src/contexts/cryptoContext.jsx [hunk 8]
```diff
@@ -12,4 +13,80 @@ const useLocalState = (key, initialValue) => {
...
+  };
+
+  const setToDsc = () => {
+    setCoinListDsc(true); 
+    console.log("setToDsc run")
+  };
+
+  const setToAsc = () => {
+    setCoinListDsc(false); 
+    console.log("setToAsc run")
```
Comment (yoloOnTheBattlefield) [inline src/contexts/cryptoContext.jsx:74]: no commented out code

File: src/pages/CoinPage.jsx [hunk 9]
```diff
@@ -87,7 +87,15 @@ const CoinPage = ({ match }) => {
             </div>
           </div>    
         </div>
-        <div className="coin-page-column-3"></div>
+        <div className="coin-page-column-3">
+          {singleCoin.market_data && <div>Market Cap: ${(singleCoin.market_data.market_cap.usd/1000000000).toFixed(2)}B&nbsp;&nbsp;{singleCoin.market_data.price_change_percentage_24h.toFixed(2)}%</div>}
+          {singleCoin.market_data && <div>Fully Diluted Valuation: ${(singleCoin.market_data.fully_diluted_valuation.usd/1000000000).toFixed(2)}B</div>}
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage.jsx:92]: make a function to format numbers

File: src/pages/Coins.jsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import React, {useState, useEffect} from "react";
+import React, {useState, useEffect, useContext} from "react";
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins.jsx:1]: add spaces {  }

File: src/pages/Coins.jsx [hunk 11]
```diff
@@ -7,8 +7,9 @@ import LineChart from "../components/LineChart"
 import BarChart from "../components/BarChart"
 import LineChartIndividualCoin from "../components/LineChartIndividualCoin"
 import '../App.css'
-import api from "../api";
+//import api from "../api";
 import InfiniteScroll from 'react-infinite-scroll-component'
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins.jsx:11]: absolute imports at the top

File: src/pages/Coins.jsx [hunk 12]
```diff
@@ -104,7 +98,7 @@ function Coins() {
       setHasError(true); 
       setisLoading(false);
     }
-  };
+  };*/
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins.jsx:101]: get rid of this

File: src/pages/Coins.jsx [hunk 13]
```diff
@@ -167,13 +161,13 @@ function Coins() {
...
 
-  const setToDsc = () => {
+  /*const setToDsc = () => {
     setCoinListDsc(true); 
   };
 
   const setToAsc = () => {
     setCoinListDsc(false); 
-  };
+  };*/
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins.jsx:170]: remove this

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 1 above)
Comment (Alcladii) [reply src/App.jsx:15]: Reformatted with Prettier

File: src/App.jsx (hunk 2 above)
Comment (Alcladii) [reply src/App.jsx:25]: added new line between line 18 and 20

File: src/App.jsx (hunk 3 above)
Comment (Alcladii) [reply src/App.jsx:52]: removed extra space at line 44

File: src/App.jsx (hunk 4 above)
Comment (Alcladii) [reply src/App.jsx:60]: fixed with useParams hook

File: src/contexts/cryptoContext.jsx (hunk 5 above)
Comment (Alcladii) [reply src/contexts/cryptoContext.jsx:36]: fixed

File: src/pages/CoinPage.jsx (hunk 9 above)
Comment (Alcladii) [reply src/pages/CoinPage.jsx:92]: Fixed. Created function in cryptoContext.jsp at line 84-86

File: src/pages/Coins.jsx (hunk 10 above)
Comment (Alcladii) [reply src/pages/Coins.jsx:1]: formatted with Prettier

File: src/pages/Coins.jsx [hunk 14]
```diff
@@ -70,190 +70,250 @@ function Coins() {
...
+      setPriceVolumeChartIsLoading(false);
+      setPriceVolumeChartIsLoadingHasError(false);
       setNumOfDays(numOfDays);
       setPriceList(data.prices);
       setVolumeList(data.total_volumes);
     } catch (err) {
-      console.log("error getting price and volume")
-      //setHasError(true);
-      //setIsLoading(false);
+      console.log("error getting price and volume");
```
Comment (yoloOnTheBattlefield) [inline src/pages/Coins.jsx:110]: no console

File: src/pages/CoinPage.jsx [hunk 15]
```diff
@@ -1,34 +1,42 @@
...
     } catch (err) {
       //console.log("Error in Fetching Data");
-      setHasError(true)
-      setIsLoading(false)
+      setSingleCoinLoadingHasError(true);
+      setSingleCoinIsLoading(false);
     }
   };
 
   //console.log(singleCoin)
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage.jsx:35]: no console

File: src/pages/CoinPage.jsx [hunk 16]
```diff
@@ -1,34 +1,42 @@
...
+        `https://api.coingecko.com/api/v3/coins/${item}?localization=false&tickers=false&market_data=true&community_data=true&developer_data=false&sparkline=false`
       );
       //console.log("from singleCoinData.data", singleCoinData.data)
-      setIsLoading(false);
+      setSingleCoinIsLoading(false);
       setSingleCoin(singleCoinData.data);
-      setHasError(false)
+      setSingleCoinLoadingHasError(false);
     } catch (err) {
       //console.log("Error in Fetching Data");
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage.jsx:29]: no console

File: src/pages/CoinPage.jsx [hunk 17]
```diff
@@ -1,34 +1,42 @@
...
+  const getSingleCoinData = async (item) => {
     try {
-      setIsLoading(true)
+      setSingleCoin({});
+      setSingleCoinIsLoading(true);
       const singleCoinData = await axios(
-        `https://api.coingecko.com/api/v3/coins/${match.params.coinId}?localization=false&tickers=false&market_data=true&community_data=true&developer_data=false&sparkline=false`
+        `https://api.coingecko.com/api/v3/coins/${item}?localization=false&tickers=false&market_data=true&community_data=true&developer_data=false&sparkline=false`
       );
       //console.log("from singleCoinData.data", singleCoinData.data)
```
Comment (yoloOnTheBattlefield) [inline src/pages/CoinPage.jsx:24]: no console

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**